File: tests/perf/test-array-write.lua
function test()
    local arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }
    for i=1,1e7 do
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
        arr[8] = 234
    end
end

test()

File: tests/perf/test-array-write.pl
sub test {
    my @arr = ( 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 );
    my $i;
    my $ign;

    for ($i = 0; $i < 1e7; $i++) {
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
        $arr[7] = 234;
    }
}

test();
